File: src/com/l2kt/gameserver/network/serverpackets/ExMultiPartyCommandChannelInfo.java
package com.l2kt.gameserver.network.serverpackets;

import com.l2kt.gameserver.model.group.CommandChannel;
import com.l2kt.gameserver.model.group.Party;

public class ExMultiPartyCommandChannelInfo extends L2GameServerPacket
{
	private final CommandChannel _channel;
	
	public ExMultiPartyCommandChannelInfo(CommandChannel channel)
	{
		_channel = channel;
	}
	
	@Override
	protected void writeImpl()
	{
		if (_channel == null)
			return;
		
		writeC(0xfe);
		writeH(0x30);
		
		writeS(_channel.getLeader().getName());
		writeD(0); // Channel loot
		writeD(_channel.getMembersCount());
		
		writeD(_channel.getParties().size());
		for (Party party : _channel.getParties())
		{
			writeS(party.getLeader().getName());
			writeD(party.getLeaderObjectId());
			writeD(party.getMembersCount());
		}
	}
}

File: src/com/l2kt/gameserver/network/serverpackets/ExMultiPartyCommandChannelInfo.kt
package com.l2kt.gameserver.network.serverpackets

import com.l2kt.gameserver.model.group.CommandChannel

class ExMultiPartyCommandChannelInfo(private val _channel: CommandChannel?) : L2GameServerPacket() {

    override fun writeImpl() {
        if (_channel == null)
            return

        writeC(0xfe)
        writeH(0x30)

        writeS(_channel.leader.name)
        writeD(0) // Channel loot
        writeD(_channel.membersCount)

        writeD(_channel.parties.size)
        for (party in _channel.parties) {
            writeS(party.leader.name)
            writeD(party.leaderObjectId)
            writeD(party.membersCount)
        }
    }
}